Python/Product/PythonTools/PythonTools/Intellisense/ProjectAnalyzer.cs

```diff
@@ -1960,6 +1974,66 @@ class ExprWalker : PythonWalker {
             return new AnalysisVariable(type, location);
         }
 
+        internal static string ExpressionForDataTip(IServiceProvider serviceProvider, ITextView view, SnapshotSpan span) {
```

Comment:
I'd rather have this be async, and do the async-to-sync adaptation at the point where it is called from the implementation of IVs* interface. Basically, as close to the boundary as possible, so most of the code is async, and gets all the benefits of that.

Also, I don't recall if we already have a generic helper to take a task and wait for it synchronously in a manner that prevents deadlocking. If we don't, you might want to port this over:

https://github.com/Microsoft/RTVS/blob/db6353d5ebdd12bcfa44b11b4a86bc8175d20777/src/Debugger/Impl/TaskExtensions.cs#L26